src/current/v23.1/cpu-profiler.md

```diff
@@ -0,0 +1,71 @@
+`server.cpu_profile.enabled` | The ON/OFF switch of the CPU profiler | `false` | 
+`server.cpu_profile.cpu_usage_combined_threshold` | The baseline value for when CPU profiles should be taken. Collect profiles from each node that meets threshold. | MAX integer, such as `9223372036854775807` | `80`
+`server.cpu_profile.interval` | The period of time after which the high-water mark resets to the baseline value | `5m0s` (5 minutes) | `1m40s` (100 seconds)
+`server.cpu_profile.duration` | The length of time a CPU profile is taken | `10s` (10 seconds) | `1s` or `2s`
+`server.cpu_profile.total_dump_size_limit` | Maximum combined disk size for preserving CPU profiles | `128 MiB` (128 Mebibytes) |
+
+### Enabling automatic CPU profile capture
+
+- Set `server.cpu_profile.enabled` to `true`
+- Set `server.cpu_profile.cpu_usage_combined_threshold` to a value between `0` and `100`.
+
+{{site.data.alerts.callout_info}}
+Only setting `server.cpu_profile.enabled` to `true` will not generate a CPU profile.
+{{site.data.alerts.end}}
+
+### Recommended values
+
+- Set `server.cpu_profile.cpu_usage_combined_threshold` to `80` for 80%.
+- Set `server.cpu_profile.duration` to a lower value, for example `1s` or `2s`. This reduces [overhead](#overhead).
+- Set `server.cpu_profile.interval` to a lower value, for example `1m40s` (1 minute 40 seconds).
+
+### High-water mark threshold 
+
+The CPU Profiler runs asynchronously in the background. After every second, the Profiler checks if the CPU usage exceeds the high-water mark threshold. If so, it captures a CPU profile. If a profile capture is already in progress, a second profile is not taken.
+
+The Profiler uses the configuration options to determine the high-water mark threshold. For example, with `duration` set to `10s` , `interval` set to `5m0s`, and `cpu_usage_combined_threshold` set to `80`:
+
+- At `time0` the CPU usage polled is 82 percent which exceeds the baseline threshold of `80`, so a `10s` profile is captured and the high-water mark threshold becomes `82`.
+- After the `10s` profile capture, the Profiler continues to check every second if the CPU usage now exceeds `82` percent.
+- At `time1` the CPU usage polled is 85 percent, another profile is taken for `10s` and the high-water mark threshold becomes `85`.
+- At `time2`, the `5m0s` interval after `time0`, the high-water mark threshold is reset to the baseline threshold of `80`.
+- The Profiler continues its every second polling and captures profiles when CPU usage exceeds the high-water mark threshold beginning at the `80` percent baseline.
+
+## Accessing CPU profiles
+
+- The CPU profiler saves the captured CPU profiles to disk in the node filesystem. The default path is `/cockroach-data/logs/pprof_dump`.
+- The only way to get the CPU profiles is to access the node filesystem.
+- Enabling the CPU profiler does **not** add CPU profiles to [debug zips](cockroach-debug-zip.html).
+
+## Overhead
```

Comment:
I recommend we add a few introductory sentences to provide context behind overhead. We should communicate explicitly that enabling the automatic CPU profile capture on a cluster will add overhead to the cluster in the form of potential increases in latency and CPU usage. The decision to enable this should be done when advised by the Cockroach Labs Support team.

Reply:
Done